cli/commands/scaffold/command.go

```diff
@@ -108,5 +110,30 @@ func GetDefaultRootFileName(ctx context.Context, opts *options.TerragruntOptions
 		return config.RecommendedParentConfigName
 	}
 
+	// Check to see if you can find the recommended parent config name first,
+	// if a user has it defined, go ahead and use it.
+	foldersToCheck := opts.MaxFoldersToCheck
+
+	dir := opts.WorkingDir
+	for dir != string(os.PathSeparator) && dir != "" && dir != "." {
+		_, err := os.Stat(filepath.Join(dir, config.RecommendedParentConfigName))
+		if err == nil {
+			return config.RecommendedParentConfigName
+		}
+
+		dir = filepath.Dir(dir)
+
+		foldersToCheck--
```

Comment:
I think descrementation and init of `foldersToCheck` can be merged in for

```

	dir := opts.WorkingDir
	for foldersToCheck := opts.MaxFoldersToCheck; dir != string(os.PathSeparator) && dir != "" && foldersToCheck > 0; foldersToCheck-- {
		if _, err := os.Stat(filepath.Join(dir, config.RecommendedParentConfigName)); err == nil {
			return config.RecommendedParentConfigName
		}
		dir = filepath.Dir(dir)
	}

```

Reply:
Is that easier to read for you?